equinox/nn/_attention.py

```diff
@@ -215,6 +237,60 @@ def __init__(
         self.use_key_bias = use_key_bias
         self.use_value_bias = use_value_bias
         self.use_output_bias = use_output_bias
+        self.autoregressive_index = StateIndex(_make_autoregressive_cache())
```

Comment:
If `state_length is None` then I think we'll hit the error message here -- so I think this will unconditionally raise?